### ISSUE TITLE
find_homography_dlt (4-point LU) hangs forever on CUDA with NaN input: torch.linalg.qr never returns

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4770]
## Describe the bug

`find_homography_dlt(..., solver="lu")` with four correspondences hangs forever on CUDA when a point is NaN. The GPU sits at 100% utilization with almost no power draw, and the Python process never returns. It cannot be interrupted short of `SIGKILL`.

This hangs the existing test `tests/geometry/test_homography.py::TestFindHomographyDLT::test_nocrash_lu[cuda-float32]`, and with it any full `KORNIA_TEST_DEVICE=cuda` run (`pixi run -e cuda test-cuda-f32`). The half-precision CUDA runs are affected too, because the gauge QR is done in float32 for half inputs.

## Cause

The minimal four-point LU path (`kornia/geometry/homography.py:261`) computes the null vector with

```python
Q, _ = torch.linalg.qr(Aw.detach().to(gauge_dtype).transpose(-2, -1), mode="complete")
```

When `points1[0, 0, 0]` is NaN, the normalized design matrix contains a structured mix of NaN, `0` and `±1`. `torch.linalg.qr` on CUDA does not return for that matrix, in `reduced`, `complete` and `r` modes alike. The CPU call returns immediately. Randomly placed NaNs (a single NaN, whole NaN rows, an all-NaN matrix) do **not** hang; the zero/NaN pattern seems to matter.

## Reproduction (pure PyTorch)

```python
import torch

nan = float("nan")
row_a = [0.0, nan] * 4   # NaN in odd columns
row_b = [nan, 0.0] * 4   # NaN in even columns
x = torch.tensor([[
    row_a, row_a, [0.0, 1.0] * 4,
    row_b, row_b, [-1.0, 0.0] * 4,
    [nan] * 8, [nan] * 8,
    torch.rand(8).tolist(),
]])  # (1, 9, 8), the matrix find_homography_dlt passes to QR on the minimal LU path
print(torch.linalg.qr(x)[1].shape)           # CPU: returns
print(torch.linalg.qr(x.cuda())[1].shape)    # CUDA: never returns, GPU at 100% utilization
```

Via kornia:

```python
import torch
from kornia.geometry.homography import find_homography_dlt

p1 = torch.rand(1, 4, 2, device="cuda")
p2 = torch.rand(1, 4, 2, device="cuda")
w = torch.ones(1, 4, device="cuda")
p1[0, 0, 0] = float("nan")
find_homography_dlt(p1, p2, w, "lu")  # hangs
```

## Expected behavior

The call returns (a NaN homography is fine, as on CPU), matching the intent of `test_nocrash_lu`.

## Possible fix

The underlying hang is a PyTorch/cuSOLVER problem, but kornia can avoid handing QR a non-finite matrix. For example, zero out non-finite entries of `Aw` before the gauge QR, and mark those batch elements invalid so that they come out as NaN, just as the solve path does now. A regression test that runs under a timeout would keep the CUDA suite from hanging silently again.

## Environment

kornia 0.9.0rc1 (`main` @ `f8449854`) | torch 2.14.0+cu130 | python 3.11.14 | OS Linux (WSL2) | CUDA 13.0 (driver 595.95), NVIDIA GeForce RTX 4090

Not checked: other torch versions (for example, the 2.5.1 CI baseline) and other GPUs.

Written by Claude (Opus 5.5) on behalf of @ducha-aiki

**Comment by ducha-aiki** [comment]
Correction/addition: the **default** call hangs as well, not only `solver="lu"` passed explicitly. The default `solver` is already `"lu"`, so

```python
find_homography_dlt(p1, p2, w)  # 4 points, one NaN -> hangs on CUDA
```

takes the same minimal-LU QR path. `TestFindHomographyDLT::test_nocrash[cuda-float32]` hits it too. It only asserts the output shape, so it reports PASSED while the QR kernel is still running asynchronously, and the hang surfaces at the next CUDA sync in the *following* test (`test_nocrash_lu`, or `test_shape[cuda-float32-2-5]` when `test_nocrash_lu` is deselected). A full CUDA run therefore needs both tests deselected until this is fixed:

```
--deselect tests/geometry/test_homography.py::TestFindHomographyDLT::test_nocrash_lu --deselect tests/geometry/test_homography.py::TestFindHomographyDLT::test_nocrash
```

Written by Claude (Opus 5.5) on behalf of @ducha-aiki
